### PR TITLE
Add advisory for elf_rs crate

### PR DESCRIPTION
**Pull request by atulkharerivos**
This adds an advisory for the elf_rs crate.

### CODE REVIEW

**Comment by pinkforest** [comment]
Has the maintainer asked about this ? I don't see any communication with the maintainer to either confirm or deny ?

**Comment by atulkharerivos** [comment]
@pinkforest I wasn't sure of the protocol, but will follow-up with the maintainer.

FWIW, here's the test program that was used with the Rust afl to reproduce the issue:

```
use elf_rs::*;
#[macro_use]
extern crate afl;

fn elf_reader(buffer: &[u8]) -> Result<(), ()> {
    let elf = Elf::from_bytes(buffer).map_err(|e| {
        println!("failed to extract elf file {:?}", e);
        ()
    })?;

    println!("{:?} header: {:?}", elf, elf.elf_header());

    for p in elf.program_header_iter() {
        println!("{:x?}", p);
    }

    for s in elf.section_header_iter() {
        println!("{:x?}", s);
    }

    if let Some(s) = elf.lookup_section(b".text") {
        println!(".test section {:?}", s);
    }
    Ok(())
}

fn main() {
    fuzz!(|data : &[u8]| {
        let _ = elf_reader(data);
    });
}
```
To reproduce, create the "in" and "out" directories, copy a small ELF binary to the "in" directory, then:

```
cargo afl fuzz -i in -o out target/debug/testelf
```

**Comment by pinkforest** [comment]
Cool no worries :) I created an issue in the repo: https://github.com/cole14/rust-elf/issues/23

Thanks a lot for the contrib in any case.

**Comment by glg-rv** [comment]
Actually this refers to https://github.com/vincenthouyi/elf_rs rather than the rust-elf create.

**Comment by cole14** [comment]
Yeah, I was just about to comment that this looks like it's fuzzing against a different crate. I'd love to see some fuzzing on the elf crate too, though :)

**Comment by atulkharerivos** [comment]
@pinkforest Just noticed what @glg-rv pointed out...the crate in question is indeed elf_rs. I will run something against rust-elf as well :).

**Comment by pinkforest** [comment]
Good catch - Thanks @glg-rv and @cole14 :+1: 

@vincenthouyi Ping

I raised https://github.com/vincenthouyi/elf_rs/issues/11

**Comment by atulkharerivos** [comment]
Apologies for the confusion. I have updated the .md to point to the correct URL (`https://github.com/vincenthouyi/elf_rs`).

File: crates/elf_rs/RUSTSEC-0000-0000.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+```toml
+[advisory]
+id = "RUSTSEC-0000-0000"
+package = "elf_rs"
+date = "2022-10-31"
+url = "https://github.com/vincenthouyi/elf_rs"
```
Comment (pinkforest) [inline crates/elf_rs/RUSTSEC-0000-0000.md:6]:
```suggestion
url = "https://github.com/vincenthouyi/elf_rs/issues/11"
```

**Comment by pinkforest** [comment]
@atulkharerivos would you mind sending a PR for elf-rs to fix it ?

https://github.com/vincenthouyi/elf_rs/issues/11#issuecomment-1300418210

Then the upstrean can roll a release we can point to in patched = []

**Comment by pinkforest** [comment]
Ok. this has been open for a while with no fix whilst the maintainer has been made aware.

Merging and will update when there is a fix / if any in the future.

**Pull request merged**